### ISSUE TITLE
Bug: Invalid value encountered in median

### ISSUE THREAD

**Issue by praftery** [CenterForTheBuiltEnvironment/mave#50]
test files 197.csv, 186.csv - seems to occur when there are any missing intervals in the data.

mave ./csv/197.csv -ts 0.7                        /usr/local/lib/python2.7/dist-packages/matplotlib/font_manager.py:273: UserWarning: Matplotlib is building the font cache using fc-list. This may take a moment.
  warnings.warn('Matplotlib is building the font cache using fc-list. This may take a moment.')
INFO     Starting mave
INFO     Assessing input file: ./csv/197.csv
INFO     No config file entered - Using default settings
INFO     Configuration used for this analysis:
{'address': '',
 'changepoints': None,
 'datetime_column_name': 'dttm_utc',
 'dayfirst': False,
 'end_frac': 1.0,
 'holiday_keys': ['USFederal'],
 'ignored_column_names': ['anomaly', 'estimated', 'timestamp'],
 'k': 10,
 'n_jobs': -1,
 'outside_db_column_name': 'OutsideDryBulbTemperature',
 'outside_dp_column_name': 'OutsideDewPointTemperature',
 'plot': False,
 'print_console': True,
 'remove_outliers': 'SingleValue',
 'save': True,
 'search_iterations': 20,
 'start_frac': 0.0,
 'target_column_name': 'value',
 'test_size': 0.7,
 'timestamp_format': '%Y-%m-%d%T%H%M',
 'use_holidays': True,
 'use_month': True,
 'use_tmy': True,
 'yearfirst': True}
INFO     No location provided
INFO     Ignoring the following columns named:['anomaly', 'estimated', 'timestamp']
INFO     Preprocessing the input file
INFO     Preprocessing started
INFO     Creating input features from datetimes
INFO     Creating other (non datetime related) input features
value
INFO     Cleaning up data - removing outliers, missing data, etc.
WARNING  Removed the following 1 outlier value(s):
[('2012-03-23 16:10:00', 50.509099999999997)]
INFO     Splitting data into pre- and post-retrofit datasets
/usr/local/lib/python2.7/dist-packages/sklearn/preprocessing/data.py:583: DeprecationWarning: Passing 1d arrays as data is deprecated in 0.17 and will raise ValueError in 0.19. Reshape your data either using X.reshape(-1, 1) if your data has a single feature or X.reshape(1, -1) if it contains a single sample.
  warnings.warn(DEPRECATION_MSG_1D, DeprecationWarning)
/usr/local/lib/python2.7/dist-packages/sklearn/preprocessing/data.py:646: DeprecationWarning: Passing 1d arrays as data is deprecated in 0.17 and will raise ValueError in 0.19. Reshape your data either using X.reshape(-1, 1) if your data has a single feature or X.reshape(1, -1) if it contains a single sample.
  warnings.warn(DEPRECATION_MSG_1D, DeprecationWarning)
INFO     Fitting models to the preretrofit data
INFO     Training hour and weekday binning models
INFO     Training random forest regressor models
INFO     Best HourWeekdayBinModel model R2 score: 0.713385516195, with parameters: {'strategy': 'mean'}
INFO     Best RandomForestRegressor model R2 score: 0.813320339418, with parameters: {'max_features': 3, 'min_samples_split': 267, 'bootstrap': False, 'max_depth': 9, 'min_samples_leaf': 52}
/usr/local/lib/python2.7/dist-packages/numpy/lib/function_base.py:3142: RuntimeWarning: Invalid value encountered in median
  RuntimeWarning)
/usr/local/lib/python2.7/dist-packages/numpy/lib/function_base.py:3403: RuntimeWarning: Invalid value encountered in median
  RuntimeWarning)
INFO     Outputting analysis results:

===== Pre-retrofit model training summary =====

=== Selected model ===
Best cross validation score on training data: 0.813320339418
Best model:
RandomForestRegressor(bootstrap=False, criterion='mse', max_depth=9,
           max_features=3, max_leaf_nodes=None, min_samples_leaf=52,
           min_samples_split=267, min_weight_fraction_leaf=0.0,
           n_estimators=10, n_jobs=1, oob_score=False, random_state=None,
           verbose=0, warm_start=False)
The relative importances of input features are:
['Minute: 0.00670161894804',
 'Hour: 0.581716661949',
 'DayOfWeek: 0.315205710804',
 'Month: 0.0585278722051',
 'Holiday: 0.0378481360934']

=== Fit to the training data ===
These error metrics represent the match between the pre-retrofit data used to train the model and the model prediction:

The model does not meet the ASHRAE Guideline 14:2002 criteria.

Negative indicates prediction > baseline (i.e. savings)
Total Biased Error: -0.000 [in original units]
Normalized Mean Bias Error: -0.000%
Mean Absolute Percent Error: 19.965%
CVRMSE: 35.767%
R2: 0.819

Distribution of normalized errors:
  minimum:    -472.156%
  10th %ile:  -32.337%
  25th %ile:  -12.932%
  median:     -1.852%
  75th %ile:  +6.923%
  90th %ile:  +18.443%
  maximum:    +75.598%

  mean:       -9.093%
  std. dev.:  45.045%
  count:      31622

===== Results =====
These results quantify the difference between the measured post-retrofit data and the predicted consumption:

Negative indicates prediction > baseline (i.e. savings)
Total Biased Error: 323136.091 [in original units]

Note: There are zero values in the baseline data rendering some typical comparison metrics meaningless.

Normalized Mean Bias Error: 34.629%
Mean Absolute Percent Error: nan%
CVRMSE: 65.305%
R2: 0.216

Distribution of normalized errors:
  minimum:    +nan%
  10th %ile:  +nan%
  25th %ile:  +nan%
  median:     +nan%
  75th %ile:  +nan%
  90th %ile:  +nan%
  maximum:    +nan%

  mean:       +nan%
  std. dev.:  nan%
  count:      73785

**Comment by praftery** [comment]
Latest commit 3458185 didn't resolve this as nans are still in the X and y arrays....

**Comment by praftery** [comment]
Interesting. The measured data in these files have a variety of issues that mean the outlier detection methods (both) fail to identify 0 as an outlier....
